docs: apps v0.23 release notes (#373)
elastisys/compliantkubernetes: docs/compliantkubernetes/release-notes.md

!!!note
    For a more detailed look check out the full [changelog](https://github.com/elastisys/compliantkubernetes-apps/blob/main/CHANGELOG.md).

### v0.23.0

Released 2022-07-06

#### Updated

> - Updated ingress-nginx, ingress-nginx controller image version v1.2.1

Reviewer: Should this line be bold as well?

Author: Yes :+1:

>   - **Breaking changes**
>     - deprecated http2_recv_timeout in favor of client_header_timeout (client-header-timeout)
>     - deprecated http2_max_field_size (http2-max-field-size) and http2_max_header_size (http2-max-header-size) in favor of large_client_header_buffers (large-client-header-buffers)
>     - deprecated http2_idle_timeout and http2_max_requests (http2-max-requests) in favor of keepalive_timeout (upstream-keepalive-timeout?) and keepalive_requests (upstream-keepalive-requests?) respectively

Reviewer: I guess these are nginx config options and thus we should be able to remove them here as they are not configurable by users
```suggestion
    - deprecated http2_idle_timeout and http2_max_requests (http2-max-requests) in favor of keepalive_timeout (upstream-keepalive-timeout?) and keepalive_requests (upstream-keepalive-requests?) respectively
```
```suggestion
  - **Breaking changes**
    - deprecated http2_recv_timeout in favor of client_header_timeout (client-header-timeout)
    - deprecated http2_max_field_size (http2-max-field-size) and http2_max_header_size (http2-max-header-size) in favor of large_client_header_buffers (large-client-header-buffers)
    - deprecated http2_idle_timeout and http2_max_requests (http2-max-requests) in favor of keepalive_timeout (upstream-keepalive-timeout?) and keepalive_requests (upstream-keepalive-requests?) respectively
```

Reviewer: I have no idea what's happening but github seems to have some problems right now.

Author: Yeah that is true. Should i remove all of them? Also that nginx is upgraded?

Reviewer 2: Maybe say that nginx removed some options and link to some other documentation of what was changed?

!!!note
    For a more detailed look check out the full [changelog](https://github.com/elastisys/compliantkubernetes-apps/blob/main/CHANGELOG.md).

### v0.23.0

Released 2022-07-06

#### Updated
- Updated ingress-nginx, ingress-nginx controller image version v1.2.1
  - **Breaking changes**
    - deprecated http2_recv_timeout in favor of client_header_timeout (client-header-timeout);
    - deprecated http2_max_field_size (http2-max-field-size) and http2_max_header_size (http2-max-header-size) in favor of large_client_header_buffers (large-client-header-buffers);
    - deprecated http2_idle_timeout and http2_max_requests (http2-max-requests) in favor of keepalive_timeout (upstream-keepalive-timeout?) and keepalive_requests (upstream-keepalive-requests?) respectively;
    - added an option to jail/chroot the nginx process, inside the controller container, is being introduced;
    - implemented an object deep inspector. The inspection is a walk through of all the spec, checking for possible attempts to escape configs.

#### Changed

> - Users are now not forced to use proxy for connecting to alertmanager but can use port-forward as well.

Reviewer: ```suggestion
- **Added support for accessing Alertmanager via port-forward**<br/>
```

#### Added

> - Add option to encrypt off-site buckets replicated with rclone sync

Reviewer: Maybe
```suggestion
- **Backups can now be encrypted before they are replicated to an off-site S3 service.**<br/>
```

> - Added metrics for field mappings and an alert that will throw an error if the fields get close to the max limit.

Reviewer: ```suggestion
- **Improved metrics and alerting for OpenSearch.**<br/> 
   When field mappings, for an index, are close to the limit, an alert will be fired.
```

#### Fixed

> - dex topologySpreadConstraints matchLabel was changed from `app: dex` to `app.kubernetes.io/name: dex` to increase stability of replica placements.

Reviewer: ```suggestion
- **The Dex deployment is now properly configured to be HA, ensuring that the Dex instances are always placed on different Kubernetes worker nodes.**<br/>
```
